route53.go

```diff
@@ -16,49 +17,79 @@ type Route53Exporter struct {
 	sess                      *session.Session
 	RecordsPerHostedZoneQuota *prometheus.Desc
 	RecordsPerHostedZoneUsage *prometheus.Desc
+	LastUpdateTime            *prometheus.Desc
 
-	logger  log.Logger
-	timeout time.Duration
+	cachedMetrics []prometheus.Metric
+	metricsMutex  *sync.Mutex
+	logger        log.Logger
+	interval      time.Duration
 }
 
-func NewRoute53Exporter(sess *session.Session, logger log.Logger, timeout time.Duration) *Route53Exporter {
+func NewRoute53Exporter(sess *session.Session, logger log.Logger, interval time.Duration) *Route53Exporter {
 
 	level.Info(logger).Log("msg", "Initializing Route53 exporter")
 
-	return &Route53Exporter{
+	exporter := &Route53Exporter{
 		sess:                      sess,
 		RecordsPerHostedZoneQuota: prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "route53_recordsperhostedzone_quota"), "Quota for maximum number of records in a Route53 hosted zone", []string{"hostedzoneid", "hostedzonename"}, nil),
 		RecordsPerHostedZoneUsage: prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "route53_recordsperhostedzone_total"), "Number of Resource records", []string{"hostedzoneid", "hostedzonename"}, nil),
+		LastUpdateTime:            prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "route53_last_updated_timestamp_seconds"), "Last time, the route53 metrics were sucessfully updated", []string{}, nil),
+		cachedMetrics:             []prometheus.Metric{},
+		metricsMutex:              &sync.Mutex{},
 		logger:                    logger,
-		timeout:                   timeout,
+		interval:                  interval,
 	}
+	go exporter.collectLoop()
```

Comment:
Makes sense. I've added it there so we don't change the interface of the Exporter. Do you think the loop should instead be started in `main.go`? This would have the benefit of making it easier to pass in a channel containing some termination signal. Right now the go-routine kinda just dies.

Reply:
Must admit, this codebase is still pretty new to me, but I think that's the best idea.